x-pack/plugins/entity_manager/server/lib/queries/index.ts

```diff
@@ -40,52 +46,86 @@ const filterCommands = ({
   start: string;
   end: string;
 }) => {
-  const commands = [
-    `WHERE ${source.timestamp_field} >= "${start}"`,
-    `WHERE ${source.timestamp_field} <= "${end}"`,
+  const filters = [
+    source.identity_fields.map((field) => `${field} IS NOT NULL`).join(' AND '),
+    ...source.filters,
   ];
 
-  source.identity_fields.forEach((field) => {
-    commands.push(`WHERE ${field} IS NOT NULL`);
-  });
-
-  source.filters.forEach((filter) => {
-    commands.push(`WHERE ${filter}`);
-  });
+  if (source.timestamp_field) {
+    filters.push(
+      `${source.timestamp_field} >= "${start}" AND ${source.timestamp_field} <= "${end}"`
+    );
+  }
 
-  return commands;
+  return filters.map((filter) => `WHERE ${filter}`).join(' | ');
 };
 
 const statsCommand = ({ source }: { source: EntitySource }) => {
-  const aggs = [
-    // default 'last_seen' attribute
-    `entity.last_seen_timestamp=MAX(${source.timestamp_field})`,
-    ...source.metadata_fields
-      .filter((field) => !source.identity_fields.some((idField) => idField === field))
-      .map((field) => `metadata.${field}=VALUES(${field})`),
-  ];
+  const aggs = source.metadata_fields
+    .filter((field) => !source.identity_fields.some((idField) => idField === field))
+    .map((field) => `${field} = VALUES(${field})`);
+
+  if (source.timestamp_field) {
+    aggs.push(`entity.last_seen_timestamp = MAX(${source.timestamp_field})`);
+  }
+
+  if (source.display_name) {
+    aggs.push(`${source.display_name} = MAX(${source.display_name})`);
```

Comment:
Is MAX the right agg to use for this?

Reply:
what we want is a way to get the latest value like `LAST()` or `TOP(field, 1, @timestamp, "desc")` but that's not available, until then I'd say there's no right agg but I'll drop a comment

Reply:
added in https://github.com/elastic/kibana/pull/202361/commits/3521f73790e283db36193eac21910bede6fe6b49